backend/modules/model_gateway/audio_processing_svc.py

```diff
@@ -0,0 +1,69 @@
+from typing import Optional
+
+import aiofiles
+import aiohttp
+
+
+class AudioProcessingSvc:
+    """
+    Async Audio Processing Service that uses Faster-Whisper Server
+    # Github: https://github.com/fedirz/faster-whisper-server
+    """
+
+    model: str
+    base_url: str
+    api_key: Optional[str] = None
+
+    def __init__(self, *, base_url: str, model: str, api_key: Optional[str] = None):
+        self.model = model
+        self.base_url = base_url
+        self.api_key = api_key
+        self.session = None
+        self.data = {
+            "model": self.model,
+            "temperature": 0.1,
+            "response_format": "json",
+            "language": "en",
+            "timestamp_granularities": "segment",
+            "stream": "true",
+        }
+
+    # allows class to be used as async context manager
+    async def __aenter__(self):
+        self.session = aiohttp.ClientSession()
+        return self
+
+    # allows class to be used as async context manager
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        if self.session:
+            await self.session.close()
+
+    async def get_transcription(self, audio_file_path: str) -> aiohttp.ClientResponse:
+        """
+        Get streaming audio transcription from Faster-Whisper Server
+        """
+        if not self.session:
+            raise RuntimeError(
+                "Session not initialized. Use 'async with' context manager."
+            )
```

Comment:
Is there really a use case to make this class itself async context manager?
Would like to keep things simpler if we can